spark3/src/main/java/org/apache/iceberg/spark/source/SparkBatchScan.java

```diff
@@ -213,10 +222,10 @@ public String description() {
     return String.format("%s [filters=%s]", table, filters);
   }
 
-  private static class ReaderFactory implements PartitionReaderFactory {
+  public static class ReaderFactory implements PartitionReaderFactory {
```

Comment:
Why public and not package-private? If we can avoid making this public, then that's a good thing.

Also, should we move these inner classes out of the `Scan` now that they are used by both `Batch` and `MicroBatchStream`?

Reply:
@rdblue - will fix the visibility from public to private-package.
I would ideally want to move these inner classes out of `Scan` - in a Different **Refactor ONLY** PR - which will keep the changes in the PR - **clearly visible** for reviewers. Did that principle makesense?!

Reply:
@rdblue - addressed the visibility feedback.

SparkBatchScan class could use a bit of Refactoring - I will follow this UP with a refactor only pr. Rationale for that is:
* changes to the codebase are easily auditable
* mix of refactor and an actual code change - will be lost - very hard for any code review to make sense out of the diff at that point.

with that, I am resolving this comment. Please feel free to reopen if you feel we must refactor in the current PR.

Reply:
Yes, follow-up issues are fine.